chore: update the workflow responsible for notifying of new TUF spec release

rdimitrov opened the pull request:
**Description of the changes being introduced by the pull request**:

The following PR deprecates the use of the existing workflow for notifying of new TUF releases and moves to one that is hosted in the https://github.com/theupdateframework/specification/ repository.

**Note:** Dependent on - https://github.com/theupdateframework/specification/pull/224

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

rdimitrov committed:
chore: update the workflow responsible for notifying of new TUF spec release

Signed-off-by: Radoslav Dimitrov <dimitrovr@vmware.com>

jku commented on .github/workflows/specification-version-check.yml:
```diff
@@ -0,0 +1,27 @@
+  # Get the latest TUF specification release and open an issue (if needed)
+  specification-bump-check:
+    needs: get-supported-tuf-version
+    uses: theupdateframework/specification/.github/workflows/check-latest-spec-version.yml@master
```
this does bring to surface a small issue with this approach: We do not want `@master` in our workflows... but on the other hand, getting dependabot updates to workflows on every spec commit sounds bad (I assume that's how it's going to work).

Hmm...

rdimitrov replied:
It's a possible drawback, indeed. Unfortunately, I think the only way to change that is to bundle it as a Github Action and create a separate repository just for it.

rdimitrov replied:
In that sense, I'll update the PR and add permissions to the job calling the workflow in order to protect the project from any vulnerabilities coming from the `specification` repo 👍

rdimitrov committed:
chore: limit the permissions for the job calling the version check workflow

Signed-off-by: Radoslav Dimitrov <dimitrovr@vmware.com>

lukpueh commented on .github/workflows/specification-version-check.yml:
```diff
@@ -0,0 +1,30 @@
+      version: ${{ steps.get-version.outputs.version }}
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+      - uses: actions/setup-python@98f2ad02fd48d057ee3b4d4f66525b231c3e52b6
```
Quoting @jku from https://github.com/theupdateframework/python-tuf/issues/2089#issuecomment-1230261767

> We should definitely set python-version to "3.x" in this workflow and everywhere else where we don't use a specific version.

Commit:
chore: misc setup-python changes in spec check job

1. update action/setup-python to latest version
2. pin major version to be used to 3.x
3. upgrade pip before using it

1 and 2 were suggested in #2089

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

lukpueh commented:
@rdimitrov: This is a friendly takeover of your PR to
1. rebase and solve conflicts
2. update action/setup-python to latest version
3. pin major version to be used to 3.x
4. upgrade pip before using it 

(3) and (4) were suggested by @jku in #2089

Given that the upstream dependent (https://github.com/theupdateframework/specification/pull/224) has been merged and permissions concerns addressed (https://github.com/theupdateframework/python-tuf/pull/2001#discussion_r868881109), I suggest this PR is good to go.

This PR also fixes parts of #2089.

jku reviewed (approved):
Looks good I suppose? Using master as the version still does not set a great example but I agree it seems safe.

Thanks!

jku commented:
oops, pip now complains: `no such option: -u`: I left #2089 open.
